scipy/signal/signaltools.py

```diff
@@ -151,10 +169,21 @@ def correlate(in1, in2, mode='full'):
         raise ValueError("Acceptable mode flags are 'valid',"
                          " 'same', or 'full'.")
 
-    if in1.ndim == in2.ndim == 0:
-        return in1 * in2
-    elif not in1.ndim == in2.ndim:
-        raise ValueError("in1 and in2 should have the same dimensionality")
+    if (in1.dtype.kind not in 'buifc' or in2.dtype.kind not in 'buifc'):
+        if method == 'fft':
+            raise ValueError('convolve can only be called with method=="fft" '
+                             'when arrays consist of numeric elements (i.e., '
+                             'int/float/etc)')
+        else:
+            method = 'direct'  # for non-numeric arrays
+    elif ((in1.dtype.kind in 'ui' or in2.dtype.kind in 'ui') and 
+          max(in1.max(), in2.max()) * max(in1.size, in2.size) > 2**52 - 1):
```

Comment:
It looks to be 2**52 everywhere. The [scipy dtype page][1] says "The default data type: float_ [aka float64]" and is said to have a 52 bit mantissa on the [scipy type page][2].

Comment added.

[2]:http://docs.scipy.org/doc/numpy/user/basics.types.html
 [1]:http://docs.scipy.org/doc/numpy/reference/arrays.dtypes.html#specifying-and-constructing-data-types